openmc/mgxs_library.py

```diff
@@ -379,200 +437,413 @@ def tabular_legendre(self, tabular_legendre):
     @num_polar.setter
     def num_polar(self, num_polar):
         # Make sure we have positive ints
-        check_value("num_polar", num_polar, Integral)
-        check_greater_than("num_polar", num_polar, 0)
+        check_value('num_polar', num_polar, Integral)
+        check_greater_than('num_polar', num_polar, 0)
         self._num_polar = num_polar
 
     @num_azimuthal.setter
     def num_azimuthal(self, num_azimuthal):
-        check_value("num_azimuthal", num_azimuthal, Integral)
-        check_greater_than("num_azimuthal", num_azimuthal, 0)
+        check_value('num_azimuthal', num_azimuthal, Integral)
+        check_greater_than('num_azimuthal', num_azimuthal, 0)
         self._num_azimuthal = num_azimuthal
 
     @total.setter
     def total(self, total):
-        if self._representation is 'isotropic':
-            shape = (self._energy_groups.num_groups,)
-        elif self._representation is 'angle':
-            shape = (self._num_polar, self._num_azimuthal,
-                     self._energy_groups.num_groups)
-        # check we have a numpy list
-        check_type("total", total, np.ndarray, expected_iter_type=Real)
-        if total.shape == shape:
-            self._total = np.copy(total)
-        else:
-            msg = 'Shape of provided total "{0}" does not match shape ' \
-                  'required, "{1}"'.format(total.shape, shape)
-            raise ValueError(msg)
+        check_type('total', total, np.ndarray, expected_iter_type=Real)
```

Comment:
It can be any Iterable. Fixing now by changing the check_type and changing the actual setting line to be `self._total =  np.array(total)` (as well as for the rest of the mgxs too)